apps/snitch_core/lib/core/domain/shipment_engine.ex

```diff
@@ -0,0 +1,152 @@
+defmodule Snitch.Domain.ShipmentEngine do
+  @moduledoc """
+  Implements functions to handle creation of the optimal shipment.
+
+  > Note 
+    No support for partially fulfilled and backordered packages.
+  """
+
+  import Aruspex.Problem, only: [new: 0, add_variable: 3, post: 4, post: 3]
+  alias Snitch.Data.Schema.Order
+  alias Aruspex.Strategy.SimulatedAnnealing
+
+  @domain [true, false]
+
+  @doc """
+  Returns a list of packages fulfilling an order
+
+  Takes a list of `packages` and an `order` struct.
+  """
+  @spec run(list, Order.t()) :: list
+  def run(packages, order) do
+    edges = create_csp(packages)
+    pckgs = append_keys(packages)
+
+    vars =
```

Comment:
Why is this a `reduce`?

Reply:
vars is a list of id's assigned to the packages `[:p1, :p2, ...]` the

Reply:
I'm just saying this could have been a `map`.